File: src/main/java/com/github/nsc/de/shake/parser/node/functions/FunctionArgumentNode.java
package com.github.nsc.de.shake.parser.node.functions;

import com.github.nsc.de.shake.parser.node.Node;
import com.github.nsc.de.shake.parser.node.VariableType;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class FunctionArgumentNode extends Node {

    private final String name;
    private final VariableType type;

    public FunctionArgumentNode(PositionMap map, String name, VariableType type) {
        super(map);
        this.name = name;
        this.type = type;
    }

    public FunctionArgumentNode(PositionMap map, String name) {
        this(map, name, VariableType.DYNAMIC);
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "FunctionArgumentNode{" +
                "name='" + name + '\'' +
                "type='" + type + '\'' +
                '}';
    }
}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/functions/FunctionArgumentNode.kt
package com.github.shakelang.shake.parser.node.functions

import com.github.shakelang.shake.parser.node.VariableType
import com.github.shakelang.shake.parser.node.Node
import com.github.shakelang.shake.util.characterinput.position.PositionMap
import kotlin.jvm.JvmOverloads

class FunctionArgumentNode @JvmOverloads constructor(
    map: PositionMap,
    val name: String,
    private val type: VariableType = VariableType.DYNAMIC
) : Node(map) {

    override fun toJson(): Map<String, *> =
        mapOf("name" to "FunctionArgumentNode", "argument_name" to name, "type" to type.toString())

}